error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> $DIR/issue-51719.rs:10:19
   |
LL | async fn foo() {}
   | -------------- this function is not `async`
...
LL |     let _gen = || foo.await;
   |                   ^^^^^^^^^ only allowed inside `async` functions and blocks

error: aborting due to previous error

